src/globus_sdk/services/timer/data.py

```diff
@@ -1,31 +1,226 @@
+    :type end: dict, optional
+
+    Example ``end`` conditions:
+
+    .. code-block:: python
+
+        # run 10 times
+        end = {"condition": "iterations", "iterations": 10}
+
+        # run until a specific date
+        end = {"condition": "time", "datetime": "2023-10-01T00:00:00Z"}
+
+    If the end condition is ``time``, then the ``datetime`` value can be expressed as a
+    python ``datetime`` type as well, e.g.
+
+    .. code-block:: python
+
+        # end in 10 days
+        end = {
+            "condition": "time",
+            "datetime": datetime.datetime.now() + datetime.timedelta(days=10),
+        }
+    """
+
+    def __init__(
+        self,
+        interval_seconds: int,
+        start: str | dt.datetime | MissingType = MISSING,
+        end: dict[str, t.Any] | MissingType = MISSING,
+    ) -> None:
+        super().__init__()
+        self["type"] = "recurring"
+        self["interval_seconds"] = interval_seconds
+        self["start"] = _format_date(start)
+        self["end"] = end
+
+        # if a datetime is given for part of the end condition, format it (and
+        # shallow-copy the end condition)
+        # primarily, this handles
+        #    end={"condition": "time", "datetime": <some-datetime>}
+        if isinstance(end, dict):
+            self["end"] = {
+                k: (_format_date(v) if isinstance(v, dt.datetime) else v)
+                for k, v in end.items()
+            }
+
+
+class OnceTimerSchedule(PayloadWrapper):
+    """
+    A helper used as part of a *timer* to define the "schedule" for the *timer*.
+
+    A ``OnceTimerSchedule`` is used to describe a *timer* which runs exactly once.
+    It may be scheduled for a time in the future.
+
+    :param datetime: The time at which to run the timer, either as an ISO 8601
+        string with timezone information, or as a ``datetime.datetime`` object.
+    :type datetime: str or datetime.datetime, optional
+    """
+
+    def __init__(
+        self,
+        datetime: str | dt.datetime | MissingType = MISSING,
+    ) -> None:
+        super().__init__()
+        self["type"] = "once"
+        self["datetime"] = _format_date(datetime)
+
+
 class TimerJob(PayloadWrapper):
     r"""
-    Class for specifying parameters used to create a job in the Timer service. Used as
-    the ``data`` argument in
-    :meth:`create_job <globus_sdk.TimerClient.create_job>`.
+    .. warning::
 
-    Timer operates through the `Globus Automate API
-    <https://docs.globus.org/globus-automation-services/>`_. Crucially, the
-    ``callback_url`` parameter should always be the URL used to run an action provider.
+        This method of specifying and creating Timers for data transfer is now
+        deprecated. Users should use ``TimerData`` instead.
 
-    .. warning::
+        ``TimerJob`` is still supported for non-transfer use-cases.
+
+    Helper for specifying a timer in the Timer service. Used as the ``data``
```

Comment:
```suggestion
    Helper for defining a timer in the Timer service. Used as the ``data``
```
```suggestion
    Helper for creating a timer in the Timer service. Used as the ``data``
```
"specifying" could mean a lot of things, a couple narrower potential revisions.

Reply:
I like `defining` for feel, but since we're using that for `TransferTimer` instantiation, I think `creating` wins as a description for the whole process (payload definition + HTTP call).
Both are better than `specifying` though.